stack.yaml

```diff
@@ -131,3 +142,11 @@ ghc-options:
   cardano-sl:            -Werror
   cardano-sl-tools:      -Werror
   cardano-sl-lwallet:    -Werror
+  cardano-sl-explorer:   -Werror
+
+# Added for explorer
+flags:
+  cardano-sl:
+    with-explorer: True
```

Comment:
> And you want me to add a flag for not building explorer?

Not for building _explorer_ but for building `cardano-sl` with `with-explorer` disabled. I. e. not building parts b. and c. But it should work only if we build explorer, otherwise the error message should be printed!

[Here](https://github.com/input-output-hk/cardano-sl/blob/49e3805ae41aacc8ed664ab3817549fb0f63e7c6/scripts/build/cardano-sl.sh#L254) you know what you are going to build. If explorer is one of the targets, `--no-explorer` is prohibited. If it's not, then `--no-explorer` is fine.

> So do you want me to change scripts/build/cardano-sl.sh so it contains explorer build by default?

Without arguments it should be everything one-by-one. `sl+` argument should also build explorer. `explorer` argument should build only explorer (and its deps).
`exporer` should be treated similar to `lwallet` and `tools`.